metric_name_builder_bench_test.go

```diff
@@ -0,0 +1,136 @@
+package prometheus
+
+import (
+	"fmt"
+	"testing"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func BenchmarkBuildMetricNames(b *testing.B) {
```

Comment:
It's better if the benchmark name matches the function being benchmarked, so readers not already familiar can map the benchmark to the function. It's a bit tricky in this case I understand because the benchmark also benchmarks `BuildCompliantMetricName`, but in any case.
```suggestion
func BenchmarkBuildMetricName(b *testing.B) {
```

Reply:
After reading the benchmarks, I think there should instead be one benchmark for `BenchmarkBuildMetricName` and one for `BenchmarkBuildMetricNames`. Mixing them will just make the benchmarks difficult comprehend and work with.